kivy/ext/mpl/backend_kivy.py

```diff
@@ -0,0 +1,386 @@
+    appropriate for your backend.
+    """
+    pass
+
+
+
+########################################################################
+#
+# The following functions and classes are for pylab and implement
+# window/figure managers, etc...
+#
+########################################################################
+
+def draw_if_interactive():
+    """
+    For image backends - is not required
+    For GUI backends - this should be overriden if drawing should be done in
+    interactive python mode
+    """
+    if matplotlib.is_interactive():
+        figManager = Gcf.get_active()
+        if figManager is not None:
+            figManager.canvas.draw_idle()
+
+def show():
+    """
+    For image backends - is not required
+    For GUI backends - show() is usually the last line of a pylab script and
+    tells the backend that it is time to draw.  In interactive mode, this may
+    be a do nothing func.  See the GTK backend for an example of how to handle
+    interactive versus batch mode
+    """
+    for manager in Gcf.get_all_fig_managers():
+        # do something to display the GUI
+        manager.show()
+
+
+def new_figure_manager(num, *args, **kwargs):
+    """
+    Create a new figure manager instance
+    """
+    # if a main-level app must be created, this (and
+    # new_figure_manager_given_figure) is the usual place to
+    # do it -- see backend_wx, backend_wxagg and backend_tkagg for
+    # examples.  Not all GUIs require explicit instantiation of a
+    # main-level app (egg backend_gtk, backend_gtkagg) for pylab
+    FigureClass = kwargs.pop('FigureClass', Figure)
+    thisFig = FigureClass(*args, **kwargs)
+    return new_figure_manager_given_figure(num, thisFig)
+
+
+def new_figure_manager_given_figure(num, figure):
+    """
+    Create a new figure manager instance for the given figure.
+    """
+    canvas = FigureCanvasKivy(figure)
+    manager = FigureManagerKivy(canvas, num)
+    return manager
+
+
+class FigureCanvasKivy(FigureCanvasKivyAgg):
+    """
+    The canvas the figure renders into.  Calls the draw and print fig
+    methods, creates the renderers, etc...
+
+    Public attribute
+
+      figure - A Figure instance
+
+    Note GUI templates will want to connect events for button presses,
+    mouse movements and key presses to functions that call the base
+    class methods button_press_event, button_release_event,
+    motion_notify_event, key_press_event, and key_release_event.  See,
+    e.g., backend_gtk.py, backend_wx.py and backend_tkagg.py
+    """
+    
+    def __init__(self, figure, **kwargs):
+        if _debug:
+            print('FigureCanvasKivy: ', figure)
+        
+        self._keyboard = Window.request_keyboard(self._keyboard_closed, self)
+        self._keyboard.bind(on_key_down = self._on_keyboard_down)
+        self._keyboard.bind(on_key_up = self._on_keyboard_up)
+        Window.bind(mouse_pos = self._on_mouse_pos)
+        Window.bind(on_close = self._on_close)
```

Comment:
As mentioned, this should probably bind to the figure being closed, rather than the whole window. Because in kivy, the window will always stay open, even if the figure dies.